### PR TITLE
Python 3.13 support

### PR DESCRIPTION
**Pull request by philipp-horstenkamp**
Python 3.13 need cython 3.
Increased cibuildwheel to its current version.
Some type conversions needed to be cast manually.

### CODE REVIEW

**Comment by philipp-horstenkamp** [comment]
closes #173

File: KDEpy/cutils_ext/cutils.pyx [hunk 1]
```diff
@@ -26,7 +26,7 @@ from libc.math cimport floor
 # for nor correctly handle negative indices
 
 # cdivision(True) -> If set to False, Cython will adjust the remainder and 
-# quotient operators C types to match those of Python ints (which differ 
+# quotient operators C types to match those of Pyethon ints (which differ
```
Comment (whitews) [inline KDEpy/cutils_ext/cutils.pyx:29]: typo?

Comment (philipp-horstenkamp) [reply KDEpy/cutils_ext/cutils.pyx]: Very much so

**Comment by philipp-horstenkamp** [comment]
It looks for me like you are building everything but the 3.13. since this isn't configured on main / master.

**Comment by whitews** [comment]
Yep, might be best to create the PR from the develop branch, but it looks like CI also needs to be updated to include 3.13 there too.

**Comment by philipp-horstenkamp** [comment]
I would recommend adding a Bild on push vor all branches for a while and revert that after merging if necessary.

Am 29. Oktober 2024 02:15:20 MEZ schrieb Scott White ***@***.***>:

**Comment by whitews** [comment]
Agreed, at least for the dev branch. Will wait for Tommy to weigh in, just wanted to move things along. The PR looks good to me. Thanks Philipp!

**Comment by philipp-horstenkamp** [comment]
It was nice to see how other people build c extensions.

**Comment by philipp-horstenkamp** [comment]
Please tag soon.
I need this for a project of mine.

**Comment by tommyod** [comment]
Yes. I think perhaps the CI/CD should build on all PRs. Just to "test" that everything runs.

I will merge this now and change a few things.

**Comment by tommyod** [comment]
Hi again. Merged and ran the deployment action now. Files are up on pypi: https://pypi.org/project/KDEpy/1.1.10/#files

If anyone wants to think about ways to simplify the GH actions pipeline with me, that would be appreciated. I see that I [started on a PR](https://github.com/tommyod/KDEpy/pull/170) on this a while ago, but never merged it.

**Comment by philipp-horstenkamp** [comment]
Please tag a new version. All user/build caches need to be invalidated otherwise.

**Comment by tommyod** [comment]
New version added :)

**Pull request merged**